protocol.go

```diff
@@ -54,6 +54,8 @@ func (p *Listener) Accept() (net.Conn, error) {
 	}
 
 	if d := p.ReadHeaderTimeout; d != 0 {
```

Comment:
We're better off moving this to only occur when the header is actually read (readHeader), that way we can decide that no proxy proto header is sent.

Reply:
Not really as that method reads from a buffer filled after reading the actual connection. I fell into this trap too before opening the PR.

Reply:
Hrm, let me do some testing. The buffer should only read once you tell it to read. It will also be at the mercy of the read deadline (which we set) so it should pass through the net.Timeout error. Unless I'm not understanding you correctly?